src/oc-id/app/controllers/profiles_controller.rb

```diff
@@ -118,4 +121,16 @@ def valid_signature?
       params[:email],
     ).valid_for?(params[:signature])
   end
+
+  def verify_user_params
+    @user = current_user
+    begin
+      raise TypeError, "Invalid First Name." unless Constants::REGEXP_NAME.match(params[:user][:first_name])
```

Comment:
Should the error message tell the user what is invalid about their input? Otherwise they might be left to guess what they need to change.